src/main/java/org/opensearch/securityanalytics/findings/FindingsService.java

```
@@ -0,0 +1,131 @@
+
+            @Override
+            public void onResponse(GetDetectorResponse getDetectorResponse) {
+                List<String> monitorIds = getDetectorResponse.getDetector().getMonitorIds();
+                if (monitorIds.size() == 0) {
+                    listener.onFailure(new IllegalArgumentException("Detector has 0 monitors"));
+                }
+
+                AtomicInteger numOfResponses = new AtomicInteger(0);
+                List<GetFindingsResponse> responses = new ArrayList<>(monitorIds.size());
+
+
+                ActionListener<GetFindingsResponse> multiGetFindingsListener = new GroupedActionListener<>(new ActionListener<>() {
+                    @Override
+                    public void onResponse(Collection<GetFindingsResponse> collection) {
+                        // Assume all responses are equal and 200
+                        RestStatus status = responses.get(0).getStatus();
+                        Integer totalFindings = 0;
+                        List<FindingWithDocs> findings = new ArrayList<>();
+                        // Merge all findings into one response
+                        for(GetFindingsResponse resp : responses) {
+                            totalFindings += resp.getTotalFindings();
+                            findings.addAll(resp.getFindings());
+                        }
+                        GetFindingsResponse masterResponse = new GetFindingsResponse(status, totalFindings, findings, detectorId);
+                        listener.onResponse(masterResponse);
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        log.error("Failed to fetch findings for detector " + detectorId, e);
+                        listener.onFailure(
+                                new IllegalArgumentException("Failed to fetch findings for detector" + detectorId)
```

let's throw a 500 exception or a SAP wrapped exception here.

This is not illegal argument as user hasn't passed any illegal/incorrect argument.

Also can we verify how many have succeeded or failed instead. Since this is a list api if a few fail we can log errors and return the rest of the findings. what say..